base/c.jl

```diff
@@ -49,7 +50,7 @@ Equivalent to the native `wchar_t` c-type ([`Int32`](@ref)).
 """
 Cwchar_t
 
-if !is_windows()
+if ccall(:jl_get_UNAME, Any, ()) !== :NT
```

Comment:
It works locally? Maybe there's something strange about evaluating it eagerly, maybe `(()->ccall(:jl_get_UNAME, Any, ()))()` would defer it to runtime or something? But then how did the const version that you had before work?